core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -3214,12 +3219,28 @@ private[spark] class CallerContext(
 
   /**
    * Set up the caller context [[context]] by invoking Hadoop CallerContext API of
-   * [[HadoopCallerContext]].
+   * [[HadoopCallerContext]], which is included in IPC calls.
+   * and the hadoop audit context, which may be included in cloud storage
+   * requests for collection in cloud service logs.
    */
-  def setCurrentContext(): Unit = if (CallerContext.callerContextEnabled) {
-    val hdfsContext = new HadoopCallerContextBuilder(context).build()
-    HadoopCallerContext.setCurrent(hdfsContext)
+  def setCurrentContext(): Unit = {
+    setCurrentContext(CallerContext.callerContextEnabled)
   }
+
+  /**
+   * Inner method to set the context.
+   * @param enabled should the thread-level contexts be updated?
+   */
+  @VisibleForTesting
+  private[util] def setCurrentContext(enabled: Boolean): Unit = {
+    if (enabled) {
+      val hdfsContext = new HadoopCallerContextBuilder(context).build()
+      HadoopCallerContext.setCurrent(hdfsContext)
+      // audit context as passed down to object stores, use prefix "spark"
+      currentAuditContext.put("spark", context)
```

Comment:
Thank you for making a PR. This one line seems to be the actual change. Did I understand correctly?

Reply:
yes! Rest of it is test related